### ISSUE TITLE
gradle dependencies unavailable, fails to build bbb-web

### ISSUE THREAD

**Issue by rcalv002** [bigbluebutton/docker#236]
Getting this error while building the images on docker compose up

It seems the site is down with forbidden errors, im not sure if its been this way before, I found an announcement on their website saying they would bring the service down in 2021 and then another announcement saying they'd leave it up as read-only. At this point its all 403 or 410 status codes.

https://blog.gradle.org/jcenter-shutdown

https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

![Snip20221030_1](https://user-images.githubusercontent.com/25310133/198902700-0dbef93f-f179-456b-8dad-2dca48cb931e.png)

**Comment by georg3k** [comment]
Having the same issue, is there a workaround?

**Comment by georg3k** [comment]
So I've found a quick workaround. You should replace jcenter repository with mavenCenter in gradle files. To do so add this:
```
# replace jcenter with mavenCenter 
RUN sed -i 's/jcenter/mavenCentral/g' /bbb-web/build.gradle && \
    sed -i 's/jcenter/mavenCentral/g' /bbb-web/pres-checker/build.gradle
```
in **mod/bbb-web/Dockerfile** right before
```
# compile bbb-web
RUN cd /bbb-web && grails assemble
```

However there seem to be another problem with mediasoup npm package v3.9.10:
```
npm ERR! cc -Isubprojects/libuv-v1.43.0/libuv.a.p -Isubprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0/include -I../../../subprojects/libuv-v1.43.0/src -fdiagnostics-color=always -DNDEBUG -std=c89 -O3 -fPIC -pthread -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112 -MD -MQ subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -MF subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o.d -o subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -c ../../../subprojects/libuv-v1.43.0/src/unix/stream.c
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c: In function ‘uv__write’:
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: error: C++ style comments are not allowed in ISO C90
npm ERR!   929 |   // XXX(jwn): this must call uv__stream_flush_write_queue(stream, n) here, since we won't generate any more events
npm ERR!       |   ^
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: note: (this will be reported only once per input file)
```

To fix this you should use the newer version of mediasoup. As a workaround add this to **mod/webrtc-sfu/Dockerfile**:
```
# Change mediasoup versions
RUN sed -i 's/3.9.10/3.10.12/g' /app/package.json && \
    sed -i 's/3.6.43/3.6.58/g' /app/package.json
```
right after
```
ENV NODE_ENV production
```
*Note that mediasoup-client v3.6.43 doesn't seem to cause any particular problems, however I decided to switch to a newer version as well.*

These changes may be unsafe, but they worked in my environment. Proper changes from BBB team are required.

**Comment by rcalv002** [comment]
Hi George!

Thanks for your comment. In fact I had run the build on Saturday without issue and then this start on Sunday when I was rebuilding. Making the the changes you recommend above it got past that step that was failing to download the dependencies but then it still fails at the end of the the build for webrtc-sfu. Did yours go through?

**Comment by georg3k** [comment]
> webrtc-sfu

Yes, the second problem in my original post occurred during webrtc-sfu container build process, specifying newer version of mediasoup helped me and compose managed to bring everything up

**Comment by rcalv002** [comment]
I already had that one at .10.3 based on another issue but I didn’t check
the client will try it and confirm thanks a bunch

**Comment by rcalv002** [comment]
Confirming the build works. Other issues of course, but the build works. Thanks George!

**Comment by georg3k** [comment]
Glad to help

**Comment by alangecker** [comment]
There have been many changes within BBB 2.6. This issue has probably also been solved. Please feel free to post again if it should still occur.
